clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -589,7 +589,8 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                         "Missing page size marker");
         report_if_false(
             shard,
-            shard->found_syscall_marker_ ==
+            shard->skipped_instrs_ ||
```

Comment:
Is this relaxation (and the one below) needed in case we skip too many instrs and don't see any syscall number marker? Did that actually happen in a test?

Intuitively it feels like these markers should be frequent enough and in most cases the skip count wouldn't be large enough as to not see any of these markers.

Reply:
Yes, both happened in these skip tests.  Although looking more closely: It looks like it's all about the filetype marker not being visible.  I changed the invariant checker to get the filetype from the stream on a skip, and removed these relaxations.

I'm still a little confused about the comments in the tests cmake for skip vs skip2: it says one will see real headers and the other won't; but neither does.  They skip 62 and 63.  Did we used to have a test trace with a chunk size of 63 or sthg?

Reply:
Going to live with that mystery.